packages/genesys-spark-components/src/components/stable/gux-popover/gux-popover.tsx

```diff
@@ -202,9 +218,13 @@ export class GuxPopover {
   }
 
   private dismiss(): void {
-    const dismissEvent = this.guxdismiss.emit();
-    if (!dismissEvent.defaultPrevented) {
-      this.isOpen = false;
+    if (this.canDismiss) {
+      this.canDismiss = false;
+      const dismissEvent = this.guxdismiss.emit();
+
+      if (!dismissEvent.defaultPrevented) {
```

Comment:
Hmm, what about adding a small value `setTimeout` to reset `canDismiss` kind of turning it into a debouncer for the event?

Reply:
i wonder would it end up simpler and more reusable to come up with a pattern to debounce the event emit function?

Reply:
Because of the way that the decorator adds the emit function, I didn't see a clean way to do that. I ended up refactoring this into a Method Decorator instead. Seems pretty clean.

Reply:
ya that is a much cleaner and reusable approach.